lib/mp-readline/readline.c

```diff
@@ -98,6 +102,38 @@ typedef struct _readline_t {
 
 STATIC readline_t rl;
 
+#if MICROPY_REPL_EMACS_KEYS
+#define FORWARD true
+#define BACKWARD false
+
+size_t get_word_cursor_pos(bool direction) {
+    char cursor_pos_step = direction == FORWARD ? 1 : -1;
+    char buf_idx_offset = direction == FORWARD ? 0 : -1;
+    size_t cursor_pos = rl.cursor_pos;
+    bool in_leading_nonalphanum = true;
+    char c;
+    while (1) {
+        c = rl.line->buf[cursor_pos + buf_idx_offset];
+        if (!(unichar_isalpha(c) || unichar_isdigit(c))) {
+            if (!in_leading_nonalphanum) {
+                break;
+            }
+        } else if (in_leading_nonalphanum) {
+            in_leading_nonalphanum = false;
+        }
+        if (direction == FORWARD) {
+            if (cursor_pos >= rl.line->len) {
+                break;
+            }
+        } else if (cursor_pos <= 0) {
+            break;
+        }
+        cursor_pos += cursor_pos_step;
+    }
+    return cursor_pos;
```

Comment:
Can probably return `cursor_pos - rl.cursor_pos` here, since all callers just want the number of chars skipped.